runtime/vm/createramclass.cpp

```diff
@@ -4005,8 +4048,14 @@ internalAllocateRAMClass(J9JavaVM *javaVM, J9ClassLoader *classLoader, RAMClassA
 			/* TODO attempt to coalesce free blocks? */
```

Comment:
line 4020->4103 should be done in a separate function since you need to do it 3 times for each segment kind.

For example

```
		for (request = requests; NULL != request; request = request->next) {
			fragmentsLeftToAllocate++;
			newSegmentSize += request->fragmentSize + request->alignment;
		}
```

That part needs to be calculated for a specific segment kind.

Reply:
```

		/* Add sizeof(UDATA) to hold the "lastAllocatedClass" pointer */
		newSegmentSize += sizeof(UDATA);
```

The following should only be done in the sub4g segment